### ISSUE TITLE
Bug: Binary Data Not Parsed Correctly in WebSocket Message Event

### ISSUE THREAD

**Issue by tomeryp** [elysiajs/elysia#247]
I've encountered a bug concerning the handling of binary data sent over a WebSocket in the Elysia.js framework. When sending binary data such as a Blob, Uint8Array, or an ArrayBuffer to the server's WebSocket endpoint, the message event handler within the .ws method does not get triggered correctly. Instead, an error is thrown indicating z.charCodeAt is not a function, which suggests the framework might be attempting to parse the message as a string.

**Steps to Reproduce:**

1. Set up a minimal Elysia.js server with a WebSocket endpoint as shown below:
```typescript
new Elysia()
.get("/", () => sendScript(clientScript))
.ws("/ws", { // does not get called
  message: (ws, message) => console.log(message, typeof message, message instanceof Buffer)
})
.listen(3000);

function clientScript() {
  const ws = new WebSocket("ws://localhost:3000/ws");
  
  // Send any type of binary data supported by the browser (Uint8Array, ArrayBuffer, Blob)
  const binaryData = new Uint8Array(1) || new ArrayBuffer(2) || new Blob([new Uint8Array(3)]);
  ws.onopen = () => ws.send(binaryData);
}

const sendScript = (script: Function) => new Response(`<html><body><script>${script.toString().replace(/^.*?{\n|}$/g,"")}</script></body></html>`, { headers: { "content-type": "text/html" } });

```

2. Navigate to http://localhost:3000 in a web browser and observe the server console for errors and logs.

**Expected Behavior:**
The websocket message event handler should be triggered, and the binary data should be logged to the server console.

**Actual Behavior:**
An error is thrown on the server-side: 

> z.charCodeAt is not a function. (In 'z.charCodeAt(0)', 'z.charCodeAt' is undefined)

**Comparison to Bun.sh:**
When implementing a similar setup directly with the Bun.sh runtime (code snippet provided below), the binary data is handled correctly and logged to the server console as expected.

```javascript
Bun.serve({
  fetch: (req, server) => server.upgrade(req) ? undefined : sendScript(clientScript),
  websocket: { // gets called and logs the buffer
    message: (ws, message) => console.log(message, typeof message, message instanceof Buffer)
  },
  port: 3000
});

function clientScript() {
  const ws = new WebSocket("ws://localhost:3000/ws");
  
  // Send any type of binary data supported by the browser (Uint8Array, ArrayBuffer, Blob)
  const binaryData = new Uint8Array(1) || new ArrayBuffer(2) || new Blob([new Uint8Array(3)]);
  ws.onopen = () => ws.send(binaryData);
}

const sendScript = (script: Function) => new Response(`<html><body><script>${script.toString().replace(/^.*?{\n|}$/g,"")}</script></body></html>`, { headers: { "content-type": "text/html" } });
```

**Environment:**

Elysia.js version: 0.7.15
Bun runtime version: 1.0.2
OS: macOS 14.0

**Comment by itpcc** [comment]
I've faced the same issue. I look it up and it seems like in `ws.parseMessage` in https://github.com/elysiajs/elysia/blob/main/src/index.ts#L2372 seems to incorrectly assume that `message`, which define as `any` 🤔,  is always is `String` and call `charCodeAt` without checking.

I'll introduce a teeny-tiny fix PR soon.

**Comment by SaltyAom** [comment]
Should be fixed in #269
